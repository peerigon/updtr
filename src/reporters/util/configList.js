import {
    USE_OPTIONS,
    UPDATE_TO_OPTIONS,
    SAVE_OPTIONS,
} from "../../constants/config";

const configNames = {
    use: "use",
    exclude: "exclude",
    test: "test command",
    registry: "registry",
    updateTo: "update to",
    save: "save",
};
const configValues = {
    exclude: list => list.join(", "),
};
const configFilter = {
    cwd: () => false,
    use: option => option !== USE_OPTIONS[0],
    exclude: list => list.length > 0,
    test: cmd => cmd !== undefined,
    registry: reg => reg !== undefined,
    updateTo: option => option !== UPDATE_TO_OPTIONS[0],
    save: option => option !== SAVE_OPTIONS[0],
};

export default function configList(config) {
    return Object.keys(config)
        .filter(key => {
            const filter = configFilter[key];
            const name = configNames[key];

            return (
                name !== undefined &&
                (filter === undefined || filter(config[key]) === true)
            );
        })
        .map(key => {
            const toString = configValues[key] || String;

            return `${ configNames[key] }: ${ toString(config[key]) }`;
        });
}
