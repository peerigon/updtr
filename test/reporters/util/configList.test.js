import configList from "../../../src/reporters/util/configList";
import {
    USE_OPTIONS,
    UPDATE_TO_OPTIONS,
    SAVE_OPTIONS,
} from "../../../src/constants/config";

describe("configList()", () => {
    describe("when an empty object is passed", () => {
        it("should return an empty array", () => {
            expect(configList({})).toEqual([]);
        });
    });
    describe("when config.cwd is specified", () => {
        it("should return an empty array", () => {
            expect(
                configList({
                    cwd: "some/path",
                })
            ).toEqual([]);
        });
    });
    describe("when config.use is", () => {
        describe(USE_OPTIONS[0], () => {
            it("should return an empty array", () => {
                expect(
                    configList({
                        use: USE_OPTIONS[0],
                    })
                ).toEqual([]);
            });
        });
        describe("something else", () => {
            it("should return the expected line", () => {
                expect(
                    configList({
                        use: "yarn",
                    })[0]
                ).toMatchSnapshot("use");
            });
        });
    });
    describe("when config.exclude is", () => {
        describe("an empty array", () => {
            it("should return an empty array", () => {
                expect(
                    configList({
                        exclude: [],
                    })
                ).toEqual([]);
            });
        });
        describe("an array with module names", () => {
            it("should return the expected line", () => {
                expect(
                    configList({
                        exclude: ["a", "b", "c"],
                    })[0]
                ).toMatchSnapshot("exclude");
            });
        });
    });
    describe("when config.updateTo is", () => {
        describe(UPDATE_TO_OPTIONS[0], () => {
            it("should return an empty array", () => {
                expect(
                    configList({
                        updateTo: UPDATE_TO_OPTIONS[0],
                    })
                ).toEqual([]);
            });
        });
        describe("something else", () => {
            it("should return the expected line", () => {
                expect(
                    configList({
                        updateTo: "non-breaking",
                    })[0]
                ).toMatchSnapshot("updateTo");
            });
        });
    });
    describe("when config.save is", () => {
        describe(SAVE_OPTIONS[0], () => {
            it("should return an empty array", () => {
                expect(
                    configList({
                        save: SAVE_OPTIONS[0],
                    })
                ).toEqual([]);
            });
        });
        describe("something else", () => {
            it("should return the expected line", () => {
                expect(
                    configList({
                        save: "exact",
                    })[0]
                ).toMatchSnapshot("save");
            });
        });
    });
    describe("when config.test is specified", () => {
        it("should return the expected line", () => {
            expect(
                configList({
                    test: "custom test command",
                })[0]
            ).toMatchSnapshot("test");
        });
    });
    describe("when config.registry is specified", () => {
        it("should return the expected line", () => {
            expect(
                configList({
                    registry: "http://example.com",
                })[0]
            ).toMatchSnapshot("registry");
        });
    });
    describe("when unknown config keys are specified", () => {
        it("should ignore them", () => {
            expect(
                configList({
                    someOtherOption: true,
                })
            ).toEqual([]);
        });
    });
});
