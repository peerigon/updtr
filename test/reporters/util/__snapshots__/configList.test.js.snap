// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exclude 1`] = `"exclude: a, b, c"`;

exports[`registry 1`] = `"registry: http://example.com"`;

exports[`save 1`] = `"save: exact"`;

exports[`test 1`] = `"test command: custom test command"`;

exports[`updateTo 1`] = `"update to: non-breaking"`;

exports[`use 1`] = `"use: yarn"`;
